CMakeLists.txt

```diff
@@ -1784,21 +1784,35 @@ if(CONFIG_BUILD_OUTPUT_INFO_HEADER)
     )
 endif()
 
-if(CONFIG_CHECK_INIT_PRIORITIES)
-  list(APPEND
-    post_build_commands
-    COMMAND ${PYTHON_EXECUTABLE} ${ZEPHYR_BASE}/scripts/build/check_init_priorities.py
-      --elf-file=${ZEPHYR_BINARY_DIR}/${KERNEL_ELF_NAME}
+if(NOT CMAKE_C_COMPILER_ID STREQUAL "ARMClang")
+  set(CHECK_INIT_PRIORITIES_INPUT
+      $<IF:$<TARGET_EXISTS:native_runner_executable>,${BYPRODUCT_KERNEL_EXE_NAME},${BYPRODUCT_KERNEL_ELF_NAME}>
+      CACHE FILEPATH "check_init_priorities input"
+  )
+  set(CHECK_INIT_PRIORITIES_COMMAND
+      ${PYTHON_EXECUTABLE} ${ZEPHYR_BASE}/scripts/build/check_init_priorities.py
+      --elf-file=${CHECK_INIT_PRIORITIES_INPUT}
+      CACHE INTERNAL "check_init_priorities command"
+  )
+  set(CHECK_INIT_PRIORITIES_DEPENDENCIES
```

Comment:
please convert to lower case, unless this is intended to be a globally used var (which I don't think)